integration/vite-manifest-invalidation-test.ts

```diff
@@ -0,0 +1,94 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+import { test } from "@playwright/test";
+import getPort from "get-port";
+
+import { createProject, viteDev, VITE_CONFIG } from "./helpers/vite.js";
+
+const files = {
+  "app/routes/_index.tsx": String.raw`
+    import { useState, useEffect } from "react";
+    import { Link } from "@remix-run/react";
+
+    export default function IndexRoute() {
+      const [mounted, setMounted] = useState(false);
+      useEffect(() => {
+        setMounted(true);
+      }, []);
+
+      return (
+        <div id="index">
+          <p data-mounted>Mounted: {mounted ? "yes" : "no"}</p>
+          <p data-hmr>HMR updated: 0</p>
+          <Link to="/other">/other</Link>
+        </div>
+      );
+    }
+  `,
+  "app/routes/other.tsx": String.raw`
+    import { useLoaderData } from "@remix-run/react";
+
+    export const loader = () => "hello";
+
+    export default function Route() {
+      const loaderData = useLoaderData();
+      return (
+        <div id="other">loaderData = {JSON.stringify(loaderData)}</div>
+      );
+    }
+  `,
+};
+
+test.describe(async () => {
+  let port: number;
+  let cwd: string;
+  let stop: () => Promise<void>;
+
+  test.beforeAll(async () => {
+    port = await getPort();
+    cwd = await createProject({
+      "vite.config.js": await VITE_CONFIG({ port }),
+      ...files,
+    });
+    stop = await viteDev({ cwd, port, debug: true });
+  });
+  test.afterAll(async () => await stop());
+
+  test("Vite / dev / invalidate manifest on route exports change", async ({
+    page,
+    context,
+    browserName,
+  }) => {
+    let pageErrors: Error[] = [];
+    page.on("pageerror", (error) => pageErrors.push(error));
+    let edit = editor(cwd);
+
+    // wait hydration to ensure initial manifest is loaded
+    await page.goto(`http://localhost:${port}/`);
+    await page.getByText("Mounted: yes").click();
+
+    // remove loader export in other page should invalidate manifest
+    await edit("app/routes/other.tsx", (contents) =>
+      contents.replace(/export const loader.*/, "")
+    );
+
+    // after browser reload, client should be aware of there's no loader
+    if (browserName === "webkit") {
+      // force new page instance for webkit.
+      // otherwise browser doesn't seem to fetch new manifest probably due to caching.
+      page = await context.newPage();
+    }
```

Comment:
I found a very odd quirk on webkit after the numerous amount of debugging.
It seems webkit is using caching for some assets between multiple `page.goto`, `reload`, etc... calls.
In this test, I'm almost certain that `/@id/__x00__virtual:browser-manifest` (aka. `browserManifestId`) was cached and webkit doesn't fetch new manifest when `page.goto` after server invalidated the module.

I hope this behavior is only playwright integration specific quirk and it doesn't happen on local usage of mac/safari.
I would appreciate if someone can verify this locally.